### PR TITLE
libnetconf4

### PR DESCRIPTION
**Pull request by Roytak**
New libnetconf2 API

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -7,15 +7,15 @@ if(${SOURCE_FORMAT_ENABLED})
...
+set(tests test_unix_socket test_client_thread test_fd_comm test_init_destroy_client test_init_destroy_server test_io test_thread_messages)
 
 # only enable PAM tests if the version of PAM is greater than 1.4
 if(LIBPAM_HAVE_CONFDIR)
-    list(APPEND tests test_pam)
+    list(APPEND tests test_auth)
 endif()
 
-set(client_tests test_client test_client_messages)
+# set(client_tests test_client test_client_messages)
```
Comment (michalvasko) [inline tests/CMakeLists.txt:18]: Meaningful client tests may be kept but should be reorganized and part of the main test suite.

**Review by michalvasko — CHANGES_REQUESTED** [review]

File: src/server_config.c [hunk 2]
```diff
@@ -0,0 +1,4648 @@
...
+        pthread_join(tid, NULL);
+    }
+
+    /* free its members */
+    free(client.name);
+
+    ch_endpt_count = client.ch_endpt_count;
+    for (i = 0; i < ch_endpt_count; i++) {
+        nc_server_config_ch_del_endpt(&client, &client.ch_endpts[i]);
+    }
```
[the remark this answers is not in the document]
Comment (Roytak) [reply src/server_config.c:1138]: I think it should be enough, but I added `pthread_mutex_lock(&client.lock);` for safety. Shouldn't be necessary though.

**Review by michalvasko — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -188,4 +188,4 @@ jobs:
         working-directory: ${{ github.workspace }}/build
         run: |
           export LSAN_OPTIONS=suppressions=${{ github.workspace }}/tests/library_lsan.supp
-          ctest --output-on-failure
+          ctest --output-on-failure
```
Comment (michalvasko) [inline .github/workflows/ci.yml:191]: Keep EOLs.

File: .github/workflows/devel-push.yml [hunk 4]
```diff
@@ -142,4 +142,4 @@ jobs:
         shell: bash
         working-directory: ${{ github.workspace }}/build
         run: bash <(curl -s https://codecov.io/bash)
-        if: ${{ matrix.config.name == 'Codecov' }}
+        if: ${{ matrix.config.name == 'Codecov' }}
```
Comment (michalvasko) [inline .github/workflows/devel-push.yml:145]: Removed EOL.

**Review by michalvasko — CHANGES_REQUESTED** [review]

**Review by michalvasko — APPROVED** [review]

**Pull request merged**